src/DssDirectDepositHub.sol

```diff
@@ -274,6 +337,17 @@ contract DssDirectDepositHub {
         emit Unwind(ilk, amount, fees);
     }
 
+    /**
+        @notice Main function for updating a D3M position.
+        Determines the current state and either winds or unwinds as necessary:
+        - MCD Caged: will unwind as much of the D3M position as possible
+        - Pool Caged/Not Culled: will unwind as much of the D3M position as possible based on the urn debt
+        - Pool Caged/Culled: will unwind as much of the D3M position as possible based on the Pool gem balance
+        - Normal Operation: Will use the D3M Plan to determine how big the desired position is and then either wind or unwind to that amount of Dai debt.
+        @dev Winding the target position will be constrained by the Ilk debt ceiling and the overall DSS debt ceiling.
```

Comment:
Also here, this seems too much info (and will be hard to maintain when/if the logic changes).